docs: admin partition and DNS clarification (#18613)
hashicorp/consul: website/content/docs/enterprise/admin-partitions.mdx

### DNS Queries

The DNS interface returns results for a single admin partition.
The query may explicitly specify the admin partition to use in the lookup.
If no admin partition is specified in the query,
the lookup uses the admin partition of the Consul agent acting as the DNS server.
Server agents always exist within the `default` admin partition.

> Client agents are configured to operate within a specific admin partition.

Author: The Consul agent doesn't initiate the DNS query. Some application team's service initiates the DNS query, the Consul agent is the thing that answers the DNS query (the DNS server).

> The DNS interface returns results for a single admin partition.
> The query may explicitly specify the admin partition to use in the lookup.
> If no admin partition is specified in the query,
> the lookup uses the admin partition of the Consul agent acting as the DNS server.
> Server agents always exist within the `default` admin partition.
> Client agents are configured to operate within a specific admin partition.

Author: Adding a suggestion here so that it's easy for me to commit the change directly from the web interface... thanks Jeff!

```suggestion
When queried, the DNS interface returns results for a single admin partition.
The query may explicitly specify the admin partition to use in the lookup.
If you do not specify an admin partition in the query,
the lookup uses the admin partition of the Consul agent that received the query.
Server agents always exist within the `default` admin partition.
Client agents are configured to operate within a specific admin partition.
```
